From: Reinhard Tartler <siretart@tauware.de>
Date: Sun, 21 Jun 2026 16:56:24 -0400
Subject: Compile against mozim 3

---
 Cargo.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Cargo.toml b/Cargo.toml
index 0e2d738..9a2c967 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -54,7 +54,7 @@ fs2 = "0.4.3"
 tokio = { version = "1.48.0", features = ["rt", "rt-multi-thread", "signal", "fs", "time"] }
 tokio-stream = { version = "0.1.17", features = ["net"] }
 tonic = "0.12.3"
-mozim = "0.2.6"
+mozim = "0.3"
 prost = "0.12.6"
 futures-channel = "0.3.31"
 futures-core = "0.3.31"
