From: Reinhard Tartler <siretart@tauware.de>
Date: Sat, 31 Aug 2024 18:28:02 -0400
Subject: Relax clap dependency

Closes: #1080118
---
 Cargo.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Cargo.toml b/Cargo.toml
index 2c63d0c..d10ed6e 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -31,7 +31,7 @@ deps-serde = ["chrono/serde", "url/serde"]
 
 [dependencies]
 anyhow = "1.0"
-clap = { version = "~4.4.8", features = ["derive"] }
+clap = { version = ">= 4.4", features = ["derive"] }
 env_logger = "0.10.1"
 ipnet = { version = "2", features = ["serde"] }
 iptables = "0.5"
