From: Reinhard Tartler <siretart@tauware.de>
Date: Sun, 21 Jun 2026 17:14:18 -0400
Subject: Compile against netlink-packet-route 0.28

---
 Cargo.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Cargo.toml b/Cargo.toml
index 9a2c967..4ea3159 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -46,7 +46,7 @@ zbus = { version = "5.12.0" }
 nix = { version = "0.30.1", features = ["net", "sched", "signal", "socket", "user"] }
 rand = "0.8.5"
 sha2 = "0.10.9"
-netlink-packet-route = "0.25.1"
+netlink-packet-route = "0.28"
 netlink-packet-core = "0.8.1"
 netlink-sys = "0.8.7"
 nftables = "0.6.3"
