From: Reinhard Tartler <siretart@tauware.de>
Date: Thu, 28 Aug 2025 16:06:00 -0400
Subject: relax depedency on log

---
 Cargo.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Cargo.toml b/Cargo.toml
index 82d18c5..e4b859e 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -39,7 +39,7 @@ env_logger = "0.11.6"
 ipnet = { version = "2.11.0", features = ["serde"] }
 iptables = "0.5.2"
 libc = "0.2.157"
-log = "0.4.27"
+log = "0.4.26"
 serde = { version = "1.0.219", features = ["derive"] }
 serde_json = "1.0.142"
 zbus = { version = "5.9.0" }
