From: Reinhard Tartler <siretart@tauware.de>
Date: Fri, 29 Aug 2025 07:31:24 -0400
Subject: Relax dependency on nftables 0.5

---
 Cargo.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Cargo.toml b/Cargo.toml
index 93c567d..118a9ad 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -49,7 +49,7 @@ sha2 = "0.10.9"
 netlink-packet-route = "0.23.0"
 netlink-packet-core = "0.7.0"
 netlink-sys = ">= 0.5"
-nftables = "0.6.3"
+nftables = "0.5.0"
 fs2 = "0.4.3"
 tokio = { version = "1.47.1", features = ["rt", "rt-multi-thread", "signal", "fs"] }
 tokio-stream = { version = "0.1.17", features = ["net"] }
