Index: netavark/src/network/core_utils.rs
===================================================================
--- netavark.orig/src/network/core_utils.rs
+++ netavark/src/network/core_utils.rs
@@ -282,8 +282,11 @@ impl CoreUtils {
     }
 }
 
-pub fn join_netns<Fd: AsFd>(fd: Fd) -> NetavarkResult<()> {
-    match sched::setns(fd, sched::CloneFlags::CLONE_NEWNET) {
+pub fn join_netns(fd: RawFd) -> NetavarkResult<()> {
+    match sched::setns(
+        unsafe { BorrowedFd::borrow_raw(fd) },
+        sched::CloneFlags::CLONE_NEWNET
+    ) {
         Ok(_) => Ok(()),
         Err(e) => Err(NetavarkError::wrap(
             "setns",
