From: Reinhard Tartler <siretart@tauware.de>
Date: Fri, 12 Jul 2024 08:36:13 -0400
Subject: relax-deps

===================================================================
---
 Cargo.toml | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/Cargo.toml b/Cargo.toml
index 4c63ded..7a61428 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -49,15 +49,15 @@ zbus = { version = "3.11.1" }
 nix = "0.26.2"
 rand = "0.8.5"
 sha2 = "0.10.6"
-netlink-packet-route = "0.15"
-netlink-packet-core = "0.5"
+netlink-packet-route = "0.17"
+netlink-packet-core = "0.7"
 fs2 = "0.4.3"
 netlink-sys = "0.8.5"
 tokio = { version = "1.27", features = ["rt", "rt-multi-thread", "signal", "fs"] }
 tokio-stream = { version = "0.1", features = ["net"] }
-tonic = "0.9"
+tonic = "0.10"
 mozim = "0.2.2"
-prost = "0.11"
+prost = "0.12"
 futures-channel="0.3.28"
 futures-core = "0.3.28"
 futures-util = "0.3.28"
@@ -68,7 +68,7 @@ tower = { version = "0.4" }
 
 [build-dependencies]
 chrono = { version = "0.4.24", default-features = false, features = ["clock"] }
-tonic-build = "0.8"
+tonic-build = "0.10"
 
 [dev-dependencies]
 once_cell = "1.17.1"
