Index: netavark/src/firewall/iptables.rs
===================================================================
--- netavark.orig/src/firewall/iptables.rs
+++ netavark/src/firewall/iptables.rs
@@ -12,7 +12,7 @@ use crate::network::types;
 use iptables;
 use iptables::IPTables;
 use log::{debug, warn};
-use zbus::blocking::Connection;
+use zbus::Connection;
 
 pub(crate) const MAX_HASH_SIZE: usize = 13;
 
@@ -249,7 +249,7 @@ fn is_firewalld_running(conn: &Connectio
 /// If possible, add a firewalld rule to allow traffic.
 /// Ignore all errors, beyond possibly logging them.
 fn add_firewalld_if_possible(net: &types::Subnet) {
-    let conn = match Connection::system() {
+    let conn = match Connection::new_system() {
         Ok(conn) => conn,
         Err(_) => return,
     };
@@ -274,7 +274,7 @@ fn add_firewalld_if_possible(net: &types
 // If possible, remove a firewalld rule to allow traffic.
 // Ignore all errors, beyond possibly logging them.
 fn rm_firewalld_if_possible(net: &types::Subnet) {
-    let conn = match Connection::system() {
+    let conn = match Connection::new_system() {
         Ok(conn) => conn,
         Err(_) => return,
     };
Index: netavark/src/firewall/mod.rs
===================================================================
--- netavark.orig/src/firewall/mod.rs
+++ netavark/src/firewall/mod.rs
@@ -4,7 +4,7 @@ use crate::network::internal_types::{
 };
 use log::{debug, info};
 use std::env;
-use zbus::blocking::Connection;
+use zbus::Connection;
 
 pub mod firewalld;
 pub mod iptables;
@@ -39,7 +39,7 @@ fn get_firewall_impl() -> NetavarkResult
         debug!("Forcibly using firewall driver {}", var);
         match var.to_lowercase().as_str() {
             "firewalld" => {
-                let conn = match Connection::system() {
+                let conn = match Connection::new_system() {
                     Ok(c) => c,
                     Err(e) => {
                         return Err(NetavarkError::wrap(
Index: netavark/src/error/mod.rs
===================================================================
--- netavark.orig/src/error/mod.rs
+++ netavark/src/error/mod.rs
@@ -70,6 +70,7 @@ pub enum NetavarkError {
     Io(std::io::Error),
 
     Dbus(zbus::Error),
+    DbusMessage(zbus::MessageError),
     DbusVariant(zbus::zvariant::Error),
 
     Sysctl(sysctl::SysctlError),
@@ -142,6 +143,7 @@ impl fmt::Display for NetavarkError {
             NetavarkError::Chain(s, e) => write!(f, "{}: {}", s, e),
             NetavarkError::Io(e) => write!(f, "IO error: {}", e),
             NetavarkError::Dbus(e) => write!(f, "DBus error: {}", e),
+            NetavarkError::DbusMessage(e) => write!(f, "DBusMessage error: {}", e),
             NetavarkError::DbusVariant(e) => write!(f, "DBus Variant Error: {}", e),
             NetavarkError::Sysctl(e) => write!(f, "Sysctl error: {}", e),
             NetavarkError::Serde(e) => write!(f, "JSON Decoding error: {}", e),
@@ -175,6 +177,12 @@ impl From<zbus::Error> for NetavarkError
     }
 }
 
+impl From<zbus::MessageError> for NetavarkError {
+    fn from(err: zbus::MessageError) -> NetavarkError {
+        NetavarkError::DbusMessage(err)
+    }
+}
+
 impl From<zbus::zvariant::Error> for NetavarkError {
     fn from(err: zbus::zvariant::Error) -> NetavarkError {
         NetavarkError::DbusVariant(err)
Index: netavark/src/firewall/firewalld.rs
===================================================================
--- netavark.orig/src/firewall/firewalld.rs
+++ netavark/src/firewall/firewalld.rs
@@ -8,7 +8,7 @@ use log::{debug, info};
 use std::collections::HashMap;
 use std::vec::Vec;
 use zbus::{
-    blocking::Connection,
+    Connection,
     zvariant::{Array, Signature, Value},
 };
 
