diff --git a/src/firewall/firewalld.rs b/src/firewall/firewalld.rs
index 32e0ff2..258f891 100644
--- a/src/firewall/firewalld.rs
+++ b/src/firewall/firewalld.rs
@@ -3,7 +3,6 @@ use crate::network::internal_types::{PortForwardConfig, TearDownNetwork, Teardow
 use crate::network::types::PortMapping;
 use crate::network::{internal_types, types};
 use core::convert::TryFrom;
-use futures::executor::block_on;
 use log::{debug, info};
 use std::collections::HashMap;
 use std::error::Error;
@@ -48,13 +47,13 @@ impl firewall::FirewallDriver for FirewallD {
 
         if need_reload {
             debug!("Reloading firewalld config to bring up zone and policy");
-            let _ = block_on(self.conn.call_method(
+            let _ = self.conn.call_method(
                 Some("org.fedoraproject.FirewallD1"),
                 "/org/fedoraproject/FirewallD1",
                 Some("org.fedoraproject.FirewallD1"),
                 "reload",
                 &(),
-            ))?;
+            )?;
         }
 
         // MUST come after the reload; otherwise the zone we made might not be
@@ -77,13 +76,13 @@ impl firewall::FirewallDriver for FirewallD {
         if let Some(subnets) = tear.config.net.subnets {
             for subnet in subnets {
                 debug!("Removing subnet {} from zone {}", subnet.subnet, ZONENAME);
-                let _ = block_on(self.conn.call_method(
+                let _ = self.conn.call_method(
                     Some("org.fedoraproject.FirewallD1"),
                     "/org/fedoraproject/FirewallD1",
                     Some("org.fedoraproject.FirewallD1.zone"),
                     "removeSource",
                     &(ZONENAME, subnet.subnet.to_string()),
-                ))?;
+                )?;
             }
         }
 
@@ -99,13 +98,13 @@ impl firewall::FirewallDriver for FirewallD {
         // I don't think there's a safer way, unfortunately.
 
         // Get the current configuration for the policy
-        let policy_config_msg = block_on(self.conn.call_method(
+        let policy_config_msg = self.conn.call_method(
             Some("org.fedoraproject.FirewallD1"),
             "/org/fedoraproject/FirewallD1",
             Some("org.fedoraproject.FirewallD1.policy"),
             "getPolicySettings",
             &(PORTPOLICYNAME),
-        ))?;
+        )?;
         let policy_config: HashMap<&str, Value> = match policy_config_msg.body() {
             Ok(m) => m,
             Err(e) => bail!(
@@ -161,13 +160,13 @@ impl firewall::FirewallDriver for FirewallD {
         new_policy_config.insert("forward_ports", &new_rules);
 
         // Send the updated configuration back to firewalld.
-        match block_on(self.conn.call_method(
+        match self.conn.call_method(
             Some("org.fedoraproject.FirewallD1"),
             "/org/fedoraproject/FirewallD1",
             Some("org.fedoraproject.FirewallD1.policy"),
             "setPolicySettings",
             &(PORTPOLICYNAME, new_policy_config),
-        )) {
+        ) {
             Ok(_) => info!(
                 "Successfully added port-forwarding rules for container {}",
                 setup_portfw.container_id
@@ -188,13 +187,13 @@ impl firewall::FirewallDriver for FirewallD {
         teardown_pf: TeardownPortForward,
     ) -> Result<(), Box<dyn Error>> {
         // Get the current configuration for the policy
-        let policy_config_msg = block_on(self.conn.call_method(
+        let policy_config_msg = self.conn.call_method(
             Some("org.fedoraproject.FirewallD1"),
             "/org/fedoraproject/FirewallD1",
             Some("org.fedoraproject.FirewallD1.policy"),
             "getPolicySettings",
             &(PORTPOLICYNAME),
-        ))?;
+        )?;
         let policy_config: HashMap<&str, Value> = match policy_config_msg.body() {
             Ok(m) => m,
             Err(e) => bail!(
@@ -259,13 +258,13 @@ impl firewall::FirewallDriver for FirewallD {
         new_policy_config.insert("forward_ports", &new_rules);
 
         // Send the updated configuration back to firewalld.
-        match block_on(self.conn.call_method(
+        match self.conn.call_method(
             Some("org.fedoraproject.FirewallD1"),
             "/org/fedoraproject/FirewallD1",
             Some("org.fedoraproject.FirewallD1.policy"),
             "setPolicySettings",
             &(PORTPOLICYNAME, new_policy_config),
-        )) {
+        ) {
             Ok(_) => info!(
                 "Successfully added port-forwarding rules for container {}",
                 teardown_pf.config.container_id
@@ -287,13 +286,13 @@ fn create_zone_if_not_exist(conn: &Connection, zone_name: &str) -> Result<bool,
     debug!("Creating firewall zone {}", zone_name);
 
     // First, double-check if the zone exists in the running config.
-    let zones_msg = block_on(conn.call_method(
+    let zones_msg = conn.call_method(
         Some("org.fedoraproject.FirewallD1"),
         "/org/fedoraproject/FirewallD1",
         Some("org.fedoraproject.FirewallD1.zone"),
         "getZones",
         &(),
-    ))?;
+    )?;
     let zones: Vec<&str> = match zones_msg.body() {
         Ok(b) => b,
         Err(e) => bail!("Error decoding DBus message for active zones: {}", e),
@@ -306,13 +305,13 @@ fn create_zone_if_not_exist(conn: &Connection, zone_name: &str) -> Result<bool,
     }
 
     // Zone is not in running config - check permanent config.
-    let perm_zones_msg = block_on(conn.call_method(
+    let perm_zones_msg = conn.call_method(
         Some("org.fedoraproject.FirewallD1"),
         "/org/fedoraproject/FirewallD1/config",
         Some("org.fedoraproject.FirewallD1.config"),
         "getZoneNames",
         &(),
-    ))?;
+    )?;
     let zones_perm: Vec<&str> = match perm_zones_msg.body() {
         Ok(b) => b,
         Err(e) => bail!("Error decoding DBus message for permanent zones: {}", e),
@@ -329,13 +328,13 @@ fn create_zone_if_not_exist(conn: &Connection, zone_name: &str) -> Result<bool,
     // errors - but I really don't want to deal with matching error strings and
     // the complexities that could entail.
     // TODO: We can add a description to the zone, should do that.
-    let _ = block_on(conn.call_method(
+    let _ = conn.call_method(
         Some("org.fedoraproject.FirewallD1"),
         "/org/fedoraproject/FirewallD1/config",
         Some("org.fedoraproject.FirewallD1.config"),
         "addZone2",
         &(zone_name, HashMap::<&str, &Value>::new()),
-    ))?;
+    )?;
 
     Ok(true)
 }
@@ -348,13 +347,13 @@ pub fn add_source_subnets_to_zone(
 ) -> Result<(), Box<dyn Error>> {
     for net in subnets {
         // Check if subnet already exists in zone
-        let subnet_zone = block_on(conn.call_method(
+        let subnet_zone = conn.call_method(
             Some("org.fedoraproject.FirewallD1"),
             "/org/fedoraproject/FirewallD1",
             Some("org.fedoraproject.FirewallD1.zone"),
             "getZoneOfSource",
             &(net.subnet.to_string()),
-        ))?;
+        )?;
         let zone_string: String = match subnet_zone.body() {
             Ok(s) => s,
             Err(e) => bail!("Error decoding DBus message for zone of subnet: {}", e),
@@ -369,13 +368,13 @@ pub fn add_source_subnets_to_zone(
             net.subnet, zone_name
         );
 
-        let _ = block_on(conn.call_method(
+        let _ = conn.call_method(
             Some("org.fedoraproject.FirewallD1"),
             "/org/fedoraproject/FirewallD1",
             Some("org.fedoraproject.FirewallD1.zone"),
             "changeZoneOfSource",
             &(zone_name, net.subnet.to_string()),
-        ))?;
+        )?;
     }
 
     Ok(())
@@ -395,13 +394,13 @@ fn add_policy_if_not_exist(
     );
 
     // Does policy exist in running policies?
-    let policies_msg = block_on(conn.call_method(
+    let policies_msg = conn.call_method(
         Some("org.fedoraproject.FirewallD1"),
         "/org/fedoraproject/FirewallD1",
         Some("org.fedoraproject.FirewallD1.policy"),
         "getPolicies",
         &(),
-    ))?;
+    )?;
     let policies: Vec<&str> = match policies_msg.body() {
         Ok(v) => v,
         Err(e) => bail!("Error decoding policy list response: {}", e),
@@ -414,13 +413,13 @@ fn add_policy_if_not_exist(
     }
 
     // Does the policy exist in permanent policies?
-    let perm_policies_msg = block_on(conn.call_method(
+    let perm_policies_msg = conn.call_method(
         Some("org.fedoraproject.FirewallD1"),
         "/org/fedoraproject/FirewallD1/config",
         Some("org.fedoraproject.FirewallD1.config"),
         "getPolicyNames",
         &(),
-    ))?;
+    )?;
     let perm_policies: Vec<&str> = match perm_policies_msg.body() {
         Ok(v) => v,
         Err(e) => bail!("Error decoding permanent policy list response: {}", e),
@@ -449,13 +448,13 @@ fn add_policy_if_not_exist(
 
     // Policy does not exist, create it.
     // Returns object path, which we don't need.
-    let _ = block_on(conn.call_method(
+    let _ = conn.call_method(
         Some("org.fedoraproject.FirewallD1"),
         "/org/fedoraproject/FirewallD1/config",
         Some("org.fedoraproject.FirewallD1.config"),
         "addPolicy",
         &(policy_name, &policy_opts),
-    ))?;
+    )?;
 
     Ok(true)
 }
diff --git a/src/firewall/iptables.rs b/src/firewall/iptables.rs
index 7fb12b9..ba7d3dd 100644
--- a/src/firewall/iptables.rs
+++ b/src/firewall/iptables.rs
@@ -9,7 +9,6 @@ use crate::network::internal_types::{
     PortForwardConfig, SetupNetwork, TearDownNetwork, TeardownPortForward,
 };
 use crate::network::types;
-use futures::executor::block_on;
 use iptables;
 use iptables::IPTables;
 use log::{debug, warn};
@@ -203,20 +202,20 @@ impl firewall::FirewallDriver for IptablesDriver {
 
 // Check if firewalld is running
 fn is_firewalld_running(conn: &Connection) -> bool {
-    block_on(conn.call_method(
+    conn.call_method(
         Some("org.freedesktop.DBus"),
         "/org/freedesktop/DBus",
         Some("org.freedesktop.DBus"),
         "GetNameOwner",
         &"org.fedoraproject.FirewallD1",
-    ))
+    )
     .is_ok()
 }
 
 // If possible, add a firewalld rule to allow traffic.
 // Ignore all errors, beyond possibly logging them.
 fn add_firewalld_if_possible(net: &types::Subnet) {
-    let conn = match block_on(Connection::system()) {
+    let conn = match Connection::new_system() {
         Ok(conn) => conn,
         Err(_) => return,
     };
@@ -241,7 +240,7 @@ fn add_firewalld_if_possible(net: &types::Subnet) {
 // If possible, remove a firewalld rule to allow traffic.
 // Ignore all errors, beyond possibly logging them.
 fn rm_firewalld_if_possible(net: &types::Subnet) {
-    let conn = match block_on(Connection::system()) {
+    let conn = match Connection::new_system() {
         Ok(conn) => conn,
         Err(_) => return,
     };
@@ -252,13 +251,13 @@ fn rm_firewalld_if_possible(net: &types::Subnet) {
         "Removing firewalld rules for IPs {}",
         net.subnet.to_string()
     );
-    match block_on(conn.call_method(
+    match conn.call_method(
         Some("org.fedoraproject.FirewallD1"),
         "/org/fedoraproject/FirewallD1",
         Some("org.fedoraproject.FirewallD1.zone"),
         "removeSource",
         &("trusted", net.subnet.to_string()),
-    )) {
+    ) {
         Ok(_) => {}
         Err(e) => warn!(
             "Error removing subnet {} from firewalld trusted zone: {}",
diff --git a/src/firewall/mod.rs b/src/firewall/mod.rs
index 5f76842..a210af8 100644
--- a/src/firewall/mod.rs
+++ b/src/firewall/mod.rs
@@ -1,7 +1,6 @@
 use crate::network::internal_types::{
     PortForwardConfig, SetupNetwork, TearDownNetwork, TeardownPortForward,
 };
-use futures::executor::block_on;
 use log::{debug, info};
 use std::env;
 use std::error::Error;
@@ -41,7 +40,7 @@ fn get_firewall_impl() -> Result<FirewallImpl, Box<dyn Error>> {
         debug!("Forcibly using firewall driver {}", var);
         match var.to_lowercase().as_str() {
             "firewalld" => {
-                let conn = match block_on(Connection::system()) {
+                let conn = match Connection::new_system() {
                     Ok(c) => c,
                     Err(e) => bail!(
                         "Error retrieving dbus connection for requested firewalld backend {}",
